prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -56,23 +86,338 @@ def _retrieve_redfish_sensor_data(self) -> List[Any]:
         Returns:
             sensors: List of dicts with details for each sensor
         """
-        try:
-            with redfish.redfish_client(
-                base_url=self.host,
-                username=self.username,
-                password=self.password,
-                timeout=self.timeout,
-                max_retry=self.max_retry,
-            ) as redfish_obj:
-                return redfish_utilities.get_sensors(redfish_obj)
-        except (
-            InvalidCredentialsError,
-            SessionCreationError,
-            ConnectionError,
-            RetriesExhaustedError,
-        ) as err:
-            logger.exception(err)
-        return []
+        sensors: List[Any] = []
+
+        redfish_obj: Optional[HttpClient] = self._get_redfish_obj()
+        if redfish_obj:
+            logger.info("Getting redfish sensor info...")
+            sensors = redfish_utilities.get_sensors(redfish_obj)
+            redfish_obj.logout()
+        return sensors
+
+    def _verify_redfish_call(self, redfish_obj: HttpClient, uri: str) -> Optional[Dict[str, Any]]:
+        """Return REST response for GET request API call on the URI.
+
+        Returns None if URI isn't present.
+        """
+        resp: RestResponse = redfish_obj.get(uri)
+        if resp.status == 200:
+            return resp.dict
+        logger.info("Not able to query from URI: %s.", uri)
+        return None
```

Comment:
**nitpick:** 

We don't need this line.

Reply:
Which line are you referring to here? Is it `return None` in which case I'm not sure what should we return. Or do you mean the `logger.info` statement? If the method which is calling `_verify_redfish_call` prints a descriptive log message in the case where we return `None`, then I guess we wouldn't need to log here.

Reply:
The line 107. It will return None even you don't add it.

Reply:
Ah okay, it completely slipped my mind that if a function doesn't return anything it effectively returns None. Thanks for the suggestion :)

Reply:
Update: Looks like I need to explicitly add the return statement. Otherwise, mypy throws an error -
`prometheus_hardware_exporter/collectors/redfish.py:98: error: Missing return statement  [return]`

Reply:
https://peps.python.org/pep-0008/#programming-recommendations
![image](https://github.com/canonical/prometheus-hardware-exporter/assets/1013484/0125d7f1-58c6-4deb-8431-bf3dcad5b0d6)